### PR TITLE
[FIX] VoIP Enable/Disable setting on CallContext/CallProvider Notifications

### PR DESCRIPTION
**Pull request by tiagoevanp**

### CODE REVIEW

File: client/providers/CallProvider/CallProvider.tsx [hunk 1]
```diff
@@ -85,13 +84,15 @@ export const CallProvider: FC = ({ children }) => {
...
 
 	useEffect(() => {
-		Notifications.onUser('callerjoined', handleQueueJoined);
-		Notifications.onUser('agentcalled', handleAgentCalled);
-		Notifications.onUser('agentconnected', handleAgentConnected);
-		Notifications.onUser('queuememberadded', handleMemberAdded);
-		Notifications.onUser('queuememberremoved', handleMemberRemoved);
-		Notifications.onUser('callabandoned', handleCallAbandon);
-		Notifications.onUser('call.callerhangup', handleCallHangup);
+		if (voipEnabled) {
```
Comment (KevLehman) [inline client/providers/CallProvider/CallProvider.tsx:87]: What if i enable voip, the events get registered, and then i disable them? Will the event listeners be destroyed? 👀

Comment (tiagoevanp) [reply client/providers/CallProvider/CallProvider.tsx:87]: ok... fixing

File: client/providers/CallProvider/CallProvider.tsx [hunk 2]
```diff
@@ -143,23 +129,58 @@ export const CallProvider: FC = ({ children }) => {
...
-		handleAgentConnected,
-		handleCallHangup,
-		handleAgentCalled,
-	]);
+		if (!voipEnabled || !user) {
+			return;
+		}
+
+		const unsubscribeFromCallerJoined = subscribeToNotifyUser(`${user._id}/callerjoined`, handleQueueJoined);
+		return unsubscribeFromCallerJoined;
```
Comment (tassoevan) [inline client/providers/CallProvider/CallProvider.tsx:137]: image

```suggestion
		return subscribeToNotifyUser(`${user._id}/callerjoined`, handleQueueJoined);
```

**Review by tassoevan — APPROVED** [review]

**Comment by KevLehman** [comment]
@tiagoevanp , not sure if this should be handled on this PR, but, if the socket is already stablished, and then you disable voip, the ws continues receiving messages 👀 (it's not finished)

Rest looks good

**Comment by tiagoevanp** [comment]
> 

Thanks for raise this @KevLehman!

We both agree on proceed with the merge and work in another time with the solution for this different problem found.

**Pull request merged**